### PR TITLE
Medical Cost Predictive Analysis

### PR DESCRIPTION
**Pull request by sitamgithub-MSIT**
# Pull Request for ML-Crate 💡

## Issue Title:  Medical Cost Predictive Analysis

- **Info about the related issue (Aim of the project)** :  The main goal of this project is to develop a machine learning model that can predict the medical charges of a person based on given features like gender, BMI index, etc.
- **Name:**  Sitam Meur
- **GitHub ID:**  sitamgithub-MSIT
- **Idenitfy yourself:**  KWOC 2023 Participant

Closes: #483

### Describe the add-ons or changes you've made 📃
The solution is implemented with technologies like Scikit-learn, Xgboost, and Catboost.
I used Kaggle datasets and split the data into test and training datasets.
Performed the necessary data pre-processing and exploratory data analysis.
Trained models like Gradient Boosting Regressor, Catboosting Regressor, Random Forest Regressor, and more.
I checked the MSE, MAE, RMSE, and R2 scores for trained models.
At the end, the accuracy of the top 3 performed models was compared by line plot, heat map, and distribution plot.

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️
The changes that I have made have been thoroughly tested in my local VS code as well as the Google collab. I also added proper requirements in the requirements.txt file for reproducibility on other machines too. Also, model graphs are added, and the accuracy table of all models is added to prove the working of the code.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by sitamgithub-MSIT** [comment]
@abhisheks008 Check out the PR, please!

**Review by abhisheks008 — APPROVED** [review]
Nicely done. Your PR is approved and ready to be merged.
@sitamgithub-MSIT
